datacube/api/core.py

```diff
@@ -526,7 +526,7 @@ def filter_jan(dataset): return dataset.time.begin.month == 1
                 dataset_predicate=dataset_predicate,
                 like=like,
                 limit=limit,
-                product=product,
+                product=product,  # type: ignore[arg-type]
```

Comment:
Does this become a kwargs parameter since the function doesn't have a parameter named product?

Even if it's wrong, I left it as is for now, the code has been there for a while.